diff --git a/src/sys.c b/src/sys.c
index fad9d1e..f1eb449 100644
--- a/src/sys.c
+++ b/src/sys.c
@@ -744,7 +744,7 @@ JL_DLLEXPORT long jl_SC_CLK_TCK(void)
 #endif
 }
 
-#ifdef _OS_OPENBSD_
+#if defined(_OS_OPENBSD_) || defined(_OS_ANDROID_)
 // Helper for jl_pathname_for_handle()
 struct dlinfo_data {
     void       *searched;
@@ -822,7 +822,16 @@ JL_DLLEXPORT const char *jl_pathname_for_handle(void *handle) JL_NOTSAFEPOINT
     dl_iterate_phdr(&dlinfo_helper, &data);
     return data.result;
 
-#else // Linux, FreeBSD, ...
+#elif defined(_OS_ANDROID_)
+    // Android/bionic doesn't have dlinfo(RTLD_DI_LINKMAP)
+    struct dlinfo_data data = {
+        .searched = handle,
+        .result = NULL,
+    };
+    dl_iterate_phdr(&dlinfo_helper, &data);
+    return data.result;
+
+#else // Linux (glibc), FreeBSD, ...
 
     struct link_map *map;
     dlinfo(handle, RTLD_DI_LINKMAP, &map);
