diff --git a/cli/loader_lib.c b/cli/loader_lib.c
index 3e15a11..93b4fa5 100644
--- a/cli/loader_lib.c
+++ b/cli/loader_lib.c
@@ -233,11 +233,16 @@ JL_DLLEXPORT const char * jl_get_libdir()
 // The path returned must be freed.
 static const char *libstdcxxprobe(void)
 {
+#if defined(__ANDROID__)
+    (void)0;
+    return NULL;
+#else
     void *handle = dlopen("libstdc++.so.6\0*", RTLD_LAZY | RTLD_LOCAL | RTLD_NOLOAD);
     if (handle != NULL)
         return NULL; // libstdc++ already loaded - nothing we can do
 
     return jl_loader_probe_system_library("libstdc++.so.6", GLIBCXX_LEAST_VERSION_SYMBOL);
+#endif
 }
 #endif
