--- a/Make.inc
+++ b/Make.inc
@@ -1541,9 +1541,6 @@
 # Warn if the user tries to build something that requires `gfortran` but they don't have it installed.
 ifeq ($(FC_VERSION),)
 ifneq ($(USE_BINARYBUILDER_OPENBLAS)$(USE_BINARYBUILDER_LIBSUITESPARSE),11)
-ifneq ($(USE_SYSTEM_OPENBLAS)$(USE_SYSTEM_LIBSUITESPARSE),11)
-$(error "Attempting to build OpenBLAS or SuiteSparse without a functioning fortran compiler!")
-endif
 endif
 endif
