--- a/src/init.c	2026-07-24 04:18:06.005393067 -0600
+++ b/src/init.c	2026-07-24 04:18:06.221393067 -0600
@@ -67,7 +67,7 @@
     // threads since it seems to return bogus values for master thread on Linux
     // and possibly OSX.
     if (!ismaster) {
-#  if defined(_OS_LINUX_) || defined(_OS_FREEBSD_)
+#  if (defined(_OS_LINUX_) && !defined(__BIONIC__)) || defined(_OS_FREEBSD_)
         pthread_attr_t attr;
 #if defined(_OS_FREEBSD_)
         pthread_attr_init(&attr);
@@ -80,6 +80,18 @@
         pthread_attr_getstack(&attr, &stackaddr, &stacksize);
         pthread_attr_destroy(&attr);
         *stack_lo = stackaddr;
+        *stack_hi = (char*)stackaddr + stacksize;
+        return;
+#  elif defined(__BIONIC__)
+        // bionic uses pthread_get_stackaddr_np/pthread_get_stacksize_np
+        void *stackaddr = pthread_get_stackaddr_np(pthread_self());
+        size_t stacksize = pthread_get_stacksize_np(pthread_self());
+        pthread_attr_t attr;
+        pthread_attr_init(&attr);
+        pthread_attr_setstack(&attr, stackaddr, stacksize);
+        pthread_attr_getstack(&attr, &stackaddr, &stacksize);
+        pthread_attr_destroy(&attr);
+        *stack_lo = stackaddr;
         *stack_hi = (char*)stackaddr + stacksize;
         return;
 #  elif defined(_OS_DARWIN_)
