diff --git a/src/signals-unix.c b/src/signals-unix.c
index 318c9a4..55d0dc0 100644
--- a/src/signals-unix.c
+++ b/src/signals-unix.c
@@ -1175,8 +1175,8 @@ static void *signal_listener(void *arg) JL_NOTSAFEPOINT
             }
         }
 
-        signal_bt_size = 0;
 #if !defined(JL_DISABLE_LIBUNWIND)
+        signal_bt_size = 0;
         if (critical) {
             do_critical_profile();
         }
@@ -1211,7 +1211,11 @@ static void *signal_listener(void *arg) JL_NOTSAFEPOINT
             // also let's make sure we're not in the middle of GC.
             jl_atomic_fetch_add(&jl_gc_disable_counter, 1);
             jl_safepoint_wait_gc(NULL);
+#if !defined(JL_DISABLE_LIBUNWIND)
             jl_exit_thread0(sig, signal_bt_data, signal_bt_size);
+#else
+            jl_exit_thread0(sig, NULL, 0);
+#endif
         }
         else if (critical) {
             // critical in this case actually means SIGINFO request
@@ -1226,10 +1230,12 @@ static void *signal_listener(void *arg) JL_NOTSAFEPOINT
 #endif
 
             jl_safe_printf("\nsignal (%d): %s\n", sig, jl_strsignal(sig));
+#if !defined(JL_DISABLE_LIBUNWIND)
             size_t i;
             for (i = 0; i < signal_bt_size; i += jl_bt_entry_size(signal_bt_data + i)) {
                 jl_fprint_bt_entry_codeloc(ios_safe_stderr, signal_bt_data + i);
             }
+#endif
             jl_safe_printf("\n");
             // Enable trace compilation to stderr with timing during profile collection
             jl_force_trace_compile_timing_enable();
