diff --git a/src/runtime_ccall.c b/src/runtime_ccall.c
index b7e3edf..d572730 100644
--- a/src/runtime_ccall.c
+++ b/src/runtime_ccall.c
@@ -9,6 +9,7 @@
 #include "julia_assert.h"
 
 #ifndef _OS_WINDOWS_
+#include <unistd.h>
 #include <sys/mman.h>
 #if defined(_OS_DARWIN_) && !defined(MAP_ANONYMOUS)
 #define MAP_ANONYMOUS MAP_ANON
@@ -152,10 +153,14 @@ JL_DLLEXPORT char *jl_format_filename(const char *output_pattern) JL_NOTSAFEPOIN
                     ios_puts(hostname, &buf);
                 }
 #ifndef _OS_WINDOWS_
+#if defined(__ANDROID__) && __ANDROID_API__ < 26
+                /* getdomainname unavailable below API 26 */
+#else
                 if (c == 'l' && getdomainname(hostname, sizeof(hostname)) == 0) {
                     hostname[sizeof(hostname) - 1] = '\0'; /* Null terminate, just to be safe. */
                     ios_puts(hostname, &buf);
                 }
+#endif
 #endif
                 break;
             case 'u':
