diff --git a/deps/libuv.mk b/deps/libuv.mk
index 62a41e1..360b383 100644
--- a/deps/libuv.mk
+++ b/deps/libuv.mk
@@ -34,6 +34,7 @@ $(LIBUV_BUILDDIR)/build-compiled: $(LIBUV_BUILDDIR)/build-configured
 	echo 1 > $@
 else
 $(LIBUV_BUILDDIR)/build-configured: $(SRCCACHE)/$(LIBUV_SRC_DIR)/source-extracted
+	cd $(SRCCACHE)/$(LIBUV_SRC_DIR) && patch -p1 -f < $(SRCDIR)/patches/libuv-android-pthread-cancel.patch
 	touch -c $(SRCCACHE)/$(LIBUV_SRC_DIR)/aclocal.m4 # touch a few files to prevent autogen from getting called
 	touch -c $(SRCCACHE)/$(LIBUV_SRC_DIR)/Makefile.in
 	touch -c $(SRCCACHE)/$(LIBUV_SRC_DIR)/configure
