--- a/src/gc-debug.c	2026-07-24 04:18:06.157393067 -0600
+++ b/src/gc-debug.c	2026-07-24 04:18:06.245393067 -0600
@@ -601,7 +601,7 @@
 #endif
 
 #ifdef GC_FINAL_STATS
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && !defined(__BIONIC__)
 #include <malloc.h> // for mallinfo
 #endif
 static double process_t0;
@@ -669,7 +669,7 @@
 
 void jl_print_gc_stats(JL_STREAM *s)
 {
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && !defined(__BIONIC__)
     malloc_stats();
 #endif
     double ptime = jl_hrtime() - process_t0;
@@ -695,7 +695,7 @@
                    p1, p1 * 100.0 / REGION1_PG_COUNT / p2,
                    p0, p0 * 100.0 / REGION0_PG_COUNT / p1);
     jl_safe_printf("image remset\t%zu entries\n", image_remset.len);
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && !defined(__BIONIC__)
     double gct = gc_num.total_time / 1e9;
     struct mallinfo mi = mallinfo();
     jl_safe_printf("malloc size\t%d MB\n", mi.uordblks / 1024 / 1024);
