--- a/src/jl_uv.c	2026-07-24 04:18:06.197393067 -0600
+++ b/src/jl_uv.c	2026-07-24 04:18:06.245393067 -0600
@@ -1067,7 +1067,7 @@
     return r;
 }
 
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && defined(TCP_QUICKACK)
 JL_DLLEXPORT int jl_tcp_quickack(uv_tcp_t *handle, int on)
 {
     int fd = (handle)->io_watcher.fd;
@@ -1212,7 +1212,7 @@
     void      *notify_data2;
 };
 
-#ifdef _OS_LINUX_
+#if defined(_OS_LINUX_) && defined(TCP_QUICKACK)
 #include <sys/syscall.h>
 #endif
 
